ai-ml/jupyterhub/addons.tf

```diff
@@ -129,23 +130,82 @@ module "eks_data_addons" {
   version = "~> 1.0" # ensure to update this to the latest/desired version
 
   oidc_provider_arn = module.eks.oidc_provider_arn
+
+  #---------------------------------------------------------------
+  # Enable GPU operator
+  #---------------------------------------------------------------
+  enable_nvidia_gpu_operator = var.jupyter_notebook_support == "gpu" ? true : false
+  nvidia_gpu_operator_helm_config = {
+    values = [templatefile("${path.module}/helm-values/nvidia-values.yaml", {})]
+  }
   #---------------------------------------------------------------
   # JupyterHub Add-on
   #---------------------------------------------------------------
   enable_jupyterhub = true
   jupyterhub_helm_config = {
-    values = [templatefile("${path.module}/helm-values/jupyterhub-values.yaml", {
-      ssl_cert_arn  = data.aws_acm_certificate.issued.arn
-      jupyterdomain = "https://${var.jupyterhub_domain}/hub/oauth_callback"
-      authorize_url = "https://${local.cognito_custom_domain}.auth.${local.region}.amazoncognito.com/oauth2/authorize"
-      token_url     = "https://${local.cognito_custom_domain}.auth.${local.region}.amazoncognito.com/oauth2/token"
-      userdata_url  = "https://${local.cognito_custom_domain}.auth.${local.region}.amazoncognito.com/oauth2/userInfo"
-      client_id     = aws_cognito_user_pool_client.user_pool_client.id
-      client_secret = aws_cognito_user_pool_client.user_pool_client.client_secret
+    values = [templatefile("${path.module}/helm-values/jupyterhub-values-${var.jupyter_hub_auth_mechanism}-${var.jupyter_notebook_support}.yaml", {
+      ssl_cert_arn                = try(data.aws_acm_certificate.issued[0].arn, "")
+      jupyterdomain               = try("https://${var.jupyterhub_domain}/hub/oauth_callback", "")
+      authorize_url               = try("https://${local.cognito_custom_domain}.auth.${local.region}.amazoncognito.com/oauth2/authorize", "")
+      token_url                   = try("https://${local.cognito_custom_domain}.auth.${local.region}.amazoncognito.com/oauth2/token", "")
+      userdata_url                = try("https://${local.cognito_custom_domain}.auth.${local.region}.amazoncognito.com/oauth2/userInfo", "")
+      client_id                   = try(aws_cognito_user_pool_client.user_pool_client[0].id, "")
+      client_secret               = try(aws_cognito_user_pool_client.user_pool_client[0].client_secret, "")
+      jupyter_single_user_sa_name = kubernetes_service_account_v1.jupyterhub_single_user_sa.metadata[0].name
     })]
   }
 }
 
+#-----------------------------------------------------------------------------------------
```

Comment:
@lusoal Could you move all the Jupyterhub related resources under a new file called `jupyterhub.tf` apart from the add-on modules? Easy to navigate what's been built for JupyterHub